standalone-metastore/src/main/thrift/hive_metastore.thrift

```diff
@@ -1887,6 +1965,38 @@ service ThriftHiveMetastore extends fb303.FacebookService
 
   WMCreateOrDropTriggerToPoolMappingResponse create_or_drop_wm_trigger_to_pool_mapping(1:WMCreateOrDropTriggerToPoolMappingRequest request)
       throws(1:AlreadyExistsException o1, 2:NoSuchObjectException o2, 3:InvalidObjectException o3, 4:MetaException o4)
+
+  // Schema calls
+  void create_ischema(1:ISchema schema) throws(1:AlreadyExistsException o1,
+        NoSuchObjectException o2, 3:MetaException o3)
+  void alter_ischema(1:string schemaName, 2:ISchema newSchema)
+        throws(1:NoSuchObjectException o1, 2:MetaException o2)
+  ISchema get_ischema(1:string schemaName) throws (1:NoSuchObjectException o1, 2:MetaException o2)
+  void drop_ischema(1:string schemaName)
+        throws(1:NoSuchObjectException o1, 2:InvalidOperationException o2, 3:MetaException o3)
+
+  void add_schema_version(1:SchemaVersion schemaVersion)
+        throws(1:AlreadyExistsException o1, 2:NoSuchObjectException o2, 3:MetaException o3)
+  SchemaVersion get_schema_version(1: string schemaName, 2: i32 version)
+        throws (1:NoSuchObjectException o1, 2:MetaException o2)
+  SchemaVersion get_schema_latest_version(1: string schemaName)
+        throws (1:NoSuchObjectException o1, 2:MetaException o2)
+  list<SchemaVersion> get_schema_all_versions(1: string schemaName)
+        throws (1:NoSuchObjectException o1, 2:MetaException o2)
+  void drop_schema_version(1: string schemaName, 2: i32 version)
+        throws(1:NoSuchObjectException o1, 2:MetaException o2)
+  FindSchemasByColsResp get_schemas_by_cols(1: FindSchemasByColsRqst rqst)
+        throws(1:MetaException o1)
+  // There is no blanket update of SchemaVersion since it is (mostly) immutable.  The only
+  // updates are the specific ones to associate a version with a serde and to change its state
+  void map_schema_version_to_serde(1: string schemaName, 2: i32 version, 3: string serdeName)
```

Comment:
Should we use Request Response objects for all the new API's ? 
Hive community seems to be agreeing in that direction - https://issues.apache.org/jira/browse/HIVE-18712?focusedCommentId=16367679&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16367679

There was also an older email thread(i think) about this topic and agreement was that future api should follow the Request/Response object pattern. (I can't find that thread anymore)

Reply:
Makes sense, especially since I discovered the hard way in the catalog work that you can't change arguments in thrift service calls in a backwards compatible way.  I'll make that change and post a new patch.